src/utilities/window.ts

```diff
@@ -0,0 +1,22 @@
+import { reduceToRecord } from './arrays'
+import { globalExists } from './global'
+import { kebabCase } from './strings'
+
+export type ComputedStyleRecord = Record<keyof CSSStyleDeclaration, string>
+
+export function getComputedStyleRecord(element: Element | undefined): ComputedStyleRecord | undefined {
+  if (!globalExists('window') || !element) {
+    return undefined
+  }
+
+  const computedStyle = window.getComputedStyle(element)
+
+  return reduceToRecord(Object.keys(computedStyle),
+    key => key as keyof CSSStyleDeclaration,
+    key => {
+      const kebab = kebabCase(key)
+
+      return computedStyle.getPropertyValue(kebab)
+    },
```

Comment:
What is this doing? Doesn't getComputedStyle already return an object with all the property styles?

Reply:
I don't full remember. I know CSSStyleDeclaration is a class and I was seeing weird things like keys both as kebab and as pascal. Either way, the ref change seems to have removed the need for it. Good catch again